Failed Download Handling

TFenby opened the pull request:
Re-opening the pull request on the development branch.

Old request: #329

Details:

Currently, it functions like this:
Disable "Post-Process Only Verified Jobs" in SABnzbd.
SABnzbd tells us when a download fails--no need for user intervention if we sent it the nzb.
On the manual post processing page, there's a checkbox to mark a manually downloaded nzb as bad.

I haven't tested it with any other nzb downloaders, and there's no way for a user to mark an auto-selected nzb as bad, since there shouldn't be a need to (with SABnzbd, at least).

I can make any changes you want to make this work how you think it should.

Commit:
Handle failed downloads

Detect and track failed nzbs and don't re-download them.

TFenby commented:
Figured I would open this on the dev branch first. Still willing to make any changes needed.

mano3m commented:
I have been running this version fo a while now and am very happy with it. I did find an issue though: if you download a season in one nzb it only marks the first episode as wanted if any of the episodes fail. It should mark all of them as failed.

Personally, I'd prefer sickbeard to always split up season nzbs if each episode has theirt own par (the code can already do this). Often only one or two episodes fail, but this causes complete failure when it combined as one nzb.

akuiraz commented:
I would also prefer sickbeard to always split up season nzbs... would be great for use when streaming nzb's using Pneumatic.

vericgar commented on sickbeard/processTV.py:
```diff
@@ -30,23 +30,115 @@
+            returnStr += logHelper(u"Not found in history, still considered Snatched: " + nzbName, logger.ERROR)
+            return returnStr
+
+        show = sql_results[0]["showid"]
```
Instead of taking just the first episode found, please loop through all the episodes returned, so they all get set back to wanted.

For example, this NZB failed for me: Transformers.Prime.S01E01-S01E08.720p.WEB-DL.DD5.1.AAC2.0.H264-BluZilla.nzb, which left episodes 2-8 still in snatched status instead of changed back to wanted.

TFenby replied:
Will do. Good catch.

TFenby replied:
What state is the folder of that download in after it fails in sab?  Could you post a directory listing?
I'm trying to find a similar download that fails on my providers.

vericgar replied:
I don't have the folder anymore. I cleaned it out after dealing with it.

There wasn't anything that seemed out of the ordinary, it just looked like a typical SAB download that had failed to repair and extract. Mostly .par and .rar.part01 and such. There was a single episode that repaired correctly, but the rest were broken.

I do still have the original e-mail from SAB saying a download failed, that includes a link to the NZB if you would like me to forward it on.

TFenby replied:
'sall good, it should be fixed now.

Commit:
Properly handle multi-episode NZBs

thezoggy commented:
while you are at it, you should go change the output format we use with sab from the old legacy text to json or xml. there is more data shared and better data just in general with those versions.

thezoggy commented:
did you test this against python 2.5?

TFenby commented:
This patch doesn't use or alter anything relating to SB's use of the sab API. If the poller patch were pulled, it would take only a tiny modification to get it to work with this.

I'll test it against 2.5 tomorrow, but I didn't use anything especially fancy, so I don't foresee any issues.
(I do use "with", but that was already imported from future)

TFenby commented:
It does work w/ 2.5, by the way.

mano3m commented:
It seems to be a bit too enthousiastic in ignoring releases. I found that the same release was posted twice, once with a password and once without. It's name on SickBeard and NZBMatrix was the same though. The script thus ignored all of them while only one out of two/three was passworded. With NZBMatrix you have the name of the post and the subject. The subjects were different. Maybe check that instead the name? Note that I am talking about True Blood S05E11 720p HDTV x264 EVOLVE if you want to test.

TFenby commented:
SB only tracks release name. If there's a bad post it should be nuked and a repacked or propered, which would give it a different release name. The one in the different group should also have been labeled "(diff group)".

TFenby commented:
Anything else blocking this from being pulled, @thezoggy?

mano3m commented:
In the case I described above the same scene release was uploaded to news servers twice  (by two different people), one failed, the other was wine. The release name was the same, nzb name was different. This scene release will not be nuked because there was nothing wrong with it. It was only uploaded to news servers wrongfully (anyone can upload...). By only tracking the release name you miss this detail and instead of marking the upload as failed, you mark the scene release as failed. If there is no other way, there is no other way, but it is not perfect....

Note that my failed download handling code is now in the develop branch of CPS, which marks an nzb as failed (this sometimes means that if two nzbs refer to the same post, it is downloaded/fails twice but at least it tries everything possible to get a working movie).

TFenby commented:
FYI - The duplicate has already been removed from nzbmatrix. The duplicate should not have been there with the same name (or password protected, for that matter) in the first place. Besides, I don't want the scope of this pull request to creep from anything other than handling when downloads fail. If one of the main devs wants to update the search code to include the group, subject, or something else, that's their call, and at that point I can update this to use that info.

mano3m commented:
Ok fair enough :) hope this gets pulled soon.

TFenby commented:
Still good to go on this.

schumi2004 commented:
I'm running a build with this PR in it but how to configure it? Currently i don't see any difference with a failed download yet. Also instaled SAB 0.7.5 and activated special config option empty_postproc but still i don't noticed anything different.

TFenby commented:
This works in the background. If a download fails, it'll say so in the log and go find another one.

schumi2004 commented:
Ah ok, thanks for explaining. I'll keep an eye on it.

christopher-schneider commented:
So is this in 497 or is it still waiting to be merged to the master?  I'm pretty new with github so I'm not sure.  If it is already in 497 do I need to do any setup in either sickbeard or sabnzbd?  Thanks.

TFenby commented:
No, it's not been pulled.

schumi2004 commented:
It's not alot of work but if you want you own copy, clone the repo and and also change the update parameters otherwise you still would have the original branche ;) Atleast that's how i did it.

christopher-schneider commented:
Thanks for the info.  I hope this does get pulled soon since I think this will be a really nice feature.  I will probably try what schumi2004 suggests in the meantime.

schumi2004 commented:
You could try the copy i made on git , it's working for me.

schumi2004 commented:
@Tolstyak 
Do you think it's possible to tweak the sabToSickBeard.py for NZBget ?

thezoggy commented:
nzbget has its own script, 

https://github.com/SynoCommunity/spksrc/blob/develop/spk/nzbget/src/nzbget-postprocess.sh
https://github.com/SynoCommunity/spksrc/blob/develop/spk/nzbget/src/nzbget-postprocess.conf

schumi2004 commented:
@thezoggy I understand but it doesn't have failed downloads in it like this https://github.com/midgetspy/Sick-Beard/tree/development/autoProcessTV

TFenby commented:
I'll check it out.

TFenby commented:
The script thezoggy linked to includes SickBeard code. Just add this after line 166:

``` python
if [ "$SickBeard" = "yes" -a "$NZBPP_CATEGORY" = "$SickBeardCategory" -a -e "$SabToSickBeard" ]; then
    # Call SickBeard's postprocessing script
    echo "[INFO] Post-Process: Running SickBeard's postprocessing script to notify it of a failed download"
    $PythonCmd $SabToSickBeard "$NZBPP_DIRECTORY" "$NZBPP_NZBFILENAME" "" "" "" "" "1" >/dev/null 2>&1
fi
```

schumi2004 commented:
Thanks will test later today.

schumi2004 commented:
Something strange is happening after adding above code. I was thinking why don't i see new episodes coming so i checked debug log for SickBeard and noticed that sabToSickBeard.py doesn't report back to SickBeard that there is a new download.
And there for nothing happens.
This is done in combination with NZBget, everything works also failed download except now postprocessing doesn't report back to SickBeard, works perfect in SABnzbd.

Edit:
Any ideas on how to remove a failed status from a previous file? Can't re-add them for testing.

TFenby commented:
The only thing that added code changed was letting SB know when a download fails. If it was working before, it should be working now. Make sure you put it after line 166 of [the original script](https://github.com/SynoCommunity/spksrc/blob/develop/spk/nzbget/src/nzbget-postprocess.sh).

schumi2004 commented:
I did and it seems to work so far except that the real script that eventually processes to SickBeard doesn't seem to work.
This script i'm talking about https://github.com/Tolstyak/Sick-Beard/blob/master/autoProcessTV/sabToSickBeard.py

Above script is called from here https://github.com/SynoCommunity/spksrc/blob/develop/spk/nzbget/src/nzbget-postprocess.sh#L321

I'f i place back old/original sabToSickBeard.py https://github.com/midgetspy/Sick-Beard/blob/master/autoProcessTV/sabToSickBeard.py then every download fails but i see in SickBeard debug log that it get's processed.

The only defference between these two are the arguments on this line https://github.com/Tolstyak/Sick-Beard/blob/master/autoProcessTV/sabToSickBeard.py#L29

Not getting it why it fails, Original sabToSickBeard processes releases but every release is marked as failed, replacement sabToSickBeard doesn't even come that far that the releases get's processed but also doesn't give any errors in log what's happening why it just stops.

TFenby commented:
nzbget doesn't show messages printed by the post processing script? It
should be complaining about there not being enough arguments passed to the
script. Add pairs of double quotes like in the modification I told you to
make earlier, but change the "1" to "0".

schumi2004 commented:
I think i have it.

Will report back later.

dVWd commented:
@schumi2004 check line 327 of your nzbget-postprocess.sh, this should be

```
$PythonCmd $SabToSickBeard "$NZBPP_DIRECTORY" "$NZBPP_NZBFILENAME" "" "" "" "" "0" >/dev/null 2>&1
```

sabToSickBeard.py now takes 7 command line arguments from SABnzbd and fails if there are less than 7, it would be better to change sabToSickBeard.py to make some arguments optional

schumi2004 commented:
@dVWd 
Made a mistake, it should be working now.

Thanks for helping all.

TFenby commented:
@dVWd: We need the seventh argument to let us know if the download failed or not, so all the others up to that point are required as well. The script is made for SAB, and nzbget is just piggybacking on it. A better solution would be to make a dedicated nzbget script, but that's not something I'm interested in doing.

schumi2004 commented:
@Tolstyak i got it working now with your SickBeard fork, everything works as expected now. 
Some other stuff i need to sort out but that's NZBget related not your fork with this patch.

schumi2004 commented:
@Tolstyak 
I updated yesterday to your latest commit but something is broken in this build

http://pastebin.com/ZXHjsNkm

TFenby commented:
Looks like a problem with the notifio notifier.

```
File "/volume1/@appstore/sickbeard/share/SickBeard/sickbeard/notifiers/notifo.py", line 69, in _sendNotifo

    logger.log(u"Unable to decode JSON: "+data, logger.ERROR)

TypeError: coercing to Unicode: need string or buffer, instance found
```

schumi2004 commented:
That would be strange since it hasn't been touched. I'll do some more testing and take a deeper look into what could be the problem.

I don't think Notifio is the only issue here.

Edit:
Not sure what is wrong but i can definitely say that commits done on development branch after 27 November has broken something.
A build from that date works flawless but newer (development) builds are broken.

Not sure if i need to create a ticket for it.

schumi2004 commented:
@Tolstyak 
After running your branch for a while i noticed something strange today.
A download was failed and couldn't be completed/processed so the failed actions came in action but it seems they weren't successful.

Deleting files leftover after failed didn't worked: Warning: Unable to remove the failed folder <path.to.release>: 13

And also the specific episode wasn't set back to wanted as described in logging, you see it in logging but when you open GUI it's still set to snatched. In a previous comment you said it's happening in the background but wouldn't it be better to change the status in the GUI to wanted as well?

http://pastebin.com/6sAzF79k

Edit:
Is this maybe the cause, the removal of def setWanted? https://github.com/Tolstyak/Sick-Beard/commit/488b6f4c642be760db9ee12390e6af1beff7ddef#L1R40 from commit Properly handle multi-episode NZBs

I tried to find out why it doesn't remove the directory when a download was failed but couldn't find it.

TFenby commented:
setWanted was removed because I changed it to set the episodes to wanted in postProcessor.py ([here](https://github.com/Tolstyak/Sick-Beard/commit/488b6f4c642be760db9ee12390e6af1beff7ddef#L0R732))

Run the following in Python and tell me the output:

``` python
import os
os.strerror(13)
```

I see the cause of the problem, though. Fix soon.

schumi2004 commented:
http://pastebin.com/Qqxur5bi

TFenby commented:
Yeah, so the user you're running SB as doesn't have permission to delete that folder. You uncovered a bug in the process, though, so that's always good.

schumi2004 commented:
So i need to fix user permissions for that specific folder or are does SB needs to be fixed?
The problem then is that i'm running both programs (SickBeard and SABnzbd) as precompiled packages from SynoCommunity and all packages are running under there own user but in same user group.
If i change user permission for incomplete folder i don't think SABnzbd can access it and vice-versa.

Thanks for helping anyway.

Edit:
users:x:100:subliminal,couchpotatoserver,tvheadend,sabnzbd,nzbget,sickbeard

TFenby commented:
The error you got was caused by something you need to fix, but the way that everything was handled after that is something I need to fix. 

You can either change the group permissions on the folder to allow group read/write/execute for the current group, or, if you don't want that whole group to have that much access, create a new group with Sab and SB users in it and assign that group to the incomplete folder with group rwx permissions.

schumi2004 commented:
I understand. 
I'll go for the complete user group then. 
Box can't be accessed from outside my network so not such a big deal but something to remember for the future.
Strange i didn't noticed this before but then again i don't have that much failed downloads yet.

Thanks.

Edit:
Ok checked the permissions and the specific folder already has full access so is the incomplete download in it.
drwxrwxrwx    3 sabnzbd  users         4096 Jan 17 20:23 incomplete

I'll ask @SABnzbd forum for a solution.

Edit2:
For anyone else in the same boat as me, this should be the solution for rights issues on incomplete folder https://forums.sabnzbd.org/viewtopic.php?t=1719#p10713

Kwebbel commented:
With the branch from Tolstyak this really works great. When can we expect it to be implemented in SB master branch?

schumi2004 commented:
@Tolstyak
Something isn't right in my opinion (or maybe it is but it's not lofic then).
Today SB recieved a few downloaded messages from failed episodes but the episode status is still on Snatched.
It would make more sense for the end user if the status would change to Wanted again if a download was failed. 
Now it's unclear what the next step would be since you don't see anything. 
Will it download a new version or maybe the same (when it's re-uploaded) on a later time?

TFenby commented:
It should set the failed episodes back to wanted, and then not download that same release in the future. You're saying it doesn't mark them as wanted? Can you post your logs?

schumi2004 commented:
It doesn't marked it as wanted yes, i can't recall it ever did btw.
this is the part from logging were it's processing it as failed
The complete log is to big and i think this is the relevant part.

http://pastebin.com/LBXnJgQ9

Kwebbel commented:
Got the same when using the development branch. When using Tolstyak's master branch it works like intended.
Strangly the log suggests something else indeed because it detects failed versions but doesnt pick another.

TFenby commented:
Should be fixed now.

Kwebbel commented:
Found another bug. In some cases when i'm searching for a specific serie the sickbeard log says it was send to sabnzb but when checking SabNZB nothing was send. Could be related to spaces in the name?

TFenby commented:
Hm. I doubt that has anything to do with this particular patch set, but I'll definitely take a look at the logs.

Kwebbel commented:
Thanks. If you're looking for an example, try to search for Arrow (S01E12)

Also found another error on the console;

Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 551, in __bootstrap_inner
    self.run()
  File "/usr/lib/python2.7/threading.py", line 504, in run
    self.__target(_self.__args, *_self.__kwargs)
  File "/opt/Sick-Beard/sickbeard/search_queue.py", line 210, in execute
    search.snatchEpisode(curResult)
  File "/opt/Sick-Beard/sickbeard/search.py", line 107, in snatchEpisode
    if result.resultType in ("nzb", "nzbdata"):
AttributeError: 'NoneType' object has no attribute 'resultType'

schumi2004 commented:
@Kwebbel
Why do you think this is related to this patch?
The files that are generating this error aren't modified by this patch.

Kwebbel commented:
@schumi2004
I'm not sure if this problem is related to the patch. I only noticed it after upgrading but it could also be another "bug".

TFenby commented:
Yeah, not related to this. You should submit a bug report on the main SB repo.

Kwebbel commented:
Running the latest development branch from Tostyak but failed series are not set to wanted.

TFenby commented:
Logs. Always send logs.

Kwebbel commented:
Busy collecting them now for you. Its pretty complicated because the backlog was running. Now doing some testing so I can provide a clean log. Should have them within a 30 minutes

Kwebbel commented:
Feb-06 14:32:45 DEBUG    CP Server Thread-6 :: Processing folder /data/DownloadData/Incomplete/The.Tudors.S03E01.720p.HDTV.X264-DIMENSION
Feb-06 14:32:45 INFO     CP Server Thread-6 :: Failed download detected: The.Tudors.S03E01.720p.HDTV.X264-DIMENSION.nzb
Feb-06 14:32:45 DEBUG    CP Server Thread-6 :: sickbeard.db: UPDATE history SET failed=1 WHERE resource=? with args [u'The.Tudors.S03E01.720p.HDTV.X264-DIMENSION']
Feb-06 14:32:45 INFO     CP Server Thread-6 :: Marked as failed. Continuing with processing episodes as failed
Feb-06 14:32:45 DEBUG    CP Server Thread-6 :: sickbeard.db: SELECT \* FROM tv_shows
Feb-06 14:32:45 DEBUG    CP Server Thread-6 :: Recursively processing a folder: **ADMIN**
Feb-06 14:32:45 DEBUG    CP Server Thread-6 :: Processing folder /data/DownloadData/Incomplete/The.Tudors.S03E01.720p.HDTV.X264-DIMENSION/**ADMIN**

After this nothing happens and status stays on snatched.

TFenby commented:
I'm trying that nzb now. In the mean time, can you post sickbeard.log from the Logs dir? Preferably to pastebin.

Kwebbel commented:
That was part of the log. Before posting the sickbeard.log I need to screen it for api key information and such. So if this is sufficient it will keep me from stripping a lot of lines :)

TFenby commented:
The relevant portions would help a whole lot.

Kwebbel commented:
Do you know a way how to remove the 'reject-because-it-failed-earlier' record for this specific episode?

TFenby commented:
```
sqlite3 sickbeard.db
UPDATE history SET failed=0 WHERE resource="The.Tudors.S03E01.720p.HDTV.X264-DIMENSION"
```

Kwebbel commented:
Thanks. http://pastebin.com/DefTTaCy

Commit:
Improve failed download handling

Let postProcessor handle finding release_name, and update history and
set episodes to wanted in the same place.

TFenby commented:
Try it now.

Kwebbel commented:
Trying it as we speak

Kwebbel commented:
I dont think this solved the problem

http://pastebin.com/YkymScnk

TFenby commented:
It didn't give a reason? Try this one, I guess.

Kwebbel commented:
Loading config from /opt/Sick-Beard/autoProcessTV/autoProcessTV.cfg
Opening URL: http://localhost:8081/home/postprocess/processEpisode?nzbName=The.Tudors.S03E01.720p.HDTV.X264-DIMENSION.nzb&failed=3&dirName=%2Fdata%2FDownloadData%2FIncomplete%2FThe.Tudors.S03E01.720p.HDTV.X264-DIMENSION.4&quiet=1
Processing folder /data/DownloadData/Incomplete/The.Tudors.S03E01.720p.HDTV.X264-DIMENSION.4
Failed download detected: (The.Tudors.S03E01.720p.HDTV.X264-DIMENSION.nzb, /data/DownloadData/Incomplete/The.Tudors.S03E01.720p.HDTV.X264-DIMENSION.4)
Processing /data/DownloadData/Incomplete/The.Tudors.S03E01.720p.HDTV.X264-DIMENSION.4 (The.Tudors.S03E01.720p.HDTV.X264-DIMENSION.nzb)
Found result in history: (79925, 3, [])
Parsed The.Tudors.S03E01.720p.HDTV.X264-DIMENSION.nzb into The Tudors - S3E1 - 720p.HDTV.X264 (DIMENSION) [ABD: False]
Checking scene exceptions for a match on The Tudors
Looking up The Tudors in the DB
Lookup successful, using tvdb id 79925
Loading show object for tvdb_id 79925
Retrieving episode object for 3x1
Release name not found. Can't mark as invalid. REPORT THIS
Processing failed for The.Tudors.S03E01.720p.HDTV.X264-DIMENSION.nzb:

Kwebbel commented:
Still the same but i'm guessing that you're looking for the "REPORT THIS" message.

TFenby commented:
Okay, so apparently it doesn't have the release name set at that point. Lame.  One sec.

Edit: Oh, duh, it's set farther down in the post processing.

TFenby commented:
Now?

Kwebbel commented:
No, internal server error

http://pastebin.com/ANbN2zL7

Kwebbel commented:
http://pastebin.com/2sELDaWy

TFenby commented:
Try that one.

Kwebbel commented:
Another internal server error:

http://pastebin.com/RaVS06kN

TFenby commented:
Annnnnd one more try.

Kwebbel commented:
The full log can be found here: http://pastebin.com/csJLvSTi

Kwebbel commented:
ah ok :) checking that now

Kwebbel commented:
No errors and the show is set to wanted again. Not sure if a new try should be triggered as well but atleast this works! Thanks alot!

TFenby commented:
Good deal. Thanks for the feedback.

Kwebbel commented:
Another issue. Another version of Arrow can not be downloaded. Although it has a failed record it still continues downloading the same version over and over again. Also it will be placed in the final destination instead of the incomplete map. 

Reason for failure:
» Repair failed, not enough repair blocks (903 short) 

http://pastebin.com/EFczfW1V

TFenby commented:
Oops, accidentally deleted my comment. Can you post the portion of the logs where it's searching for releases? After the logs you just posted would be preferred, since it will have been marked as failed at that point.

Also, would you care to test my failed-database branch? I've added a page to the web UI for managing failed downloads.

Kwebbel commented:
The strange thing is that it is placed in /data/Media/TV instead of /data/DownloadData/Incomplete. The last location should be the place stuff is downloaded into until processing. In this case Arrow will go straight into /data/Media/TV which is somehow wierd because I would expect that sickbeard moved it to there because it thought it was succesfully downloaded.

Log:
http://pastebin.com/EKg5nU6i

Could it be that ARROW is a valid SQL command which causes the query to break? Other series dont have this particular issue.

schumi2004 commented:
@Kwebbel
I think i can explain why it starts in /data/Media/TV
I think your using Sab, have you set directory /data/Media/TV for categorie series/tv or what you use for SickBeard?
If so then it skips/bypasses complete/incomplete folders.

@Tolstyak
You mean this branche, https://github.com/Tolstyak/Sick-Beard/commits/failed-database

I'll test it in a few.

Kwebbel commented:
Indeed but howcome other incompletes are placed within the Incomplete folder?

For example:

Loading config from /opt/Sick-Beard/autoProcessTV/autoProcessTV.cfg
Opening URL: http://localhost:8081/home/postprocess/processEpisode?nzbName=Boss.S01E02.720p.HDTV.DD5.1.x264-EbP.nzb&failed=3&dirName=%2Fdata%2FDownloadData%2FIncomplete%2FBoss.S01E02.720p.HDTV.DD5.1.x264-EbP&quiet=1
Processing folder /data/DownloadData/Incomplete/Boss.S01E02.720p.HDTV.DD5.1.x264-EbP
Failed download detected: (Boss.S01E02.720p.HDTV.DD5.1.x264-EbP.nzb, /data/DownloadData/Incomplete/Boss.S01E02.720p.HDTV.DD5.1.x264-EbP)
Processing /data/DownloadData/Incomplete/Boss.S01E02.720p.HDTV.DD5.1.x264-EbP (Boss.S01E02.720p.HDTV.DD5.1.x264-EbP.nzb)
Found result in history: (97221, 1, [])
Parsed Boss.S01E02.720p.HDTV.DD5.1.x264-EbP.nzb into Boss - S1E2 - 720p.HDTV.DD5.1.x264 (EbP) [ABD: False]
Checking scene exceptions for a match on Boss
Scene exception lookup got tvdb id 249361, using that
Loading show object for tvdb_id 97221
Retrieving episode object for 1x2
Marking release as bad: Boss.S01E02.720p.HDTV.DD5.1.x264-EbP
Setting episode back to wanted: A Crime in the Name of Curiosity
Deleting folder of failed download /data/DownloadData/Incomplete/Boss.S01E02.720p.HDTV.DD5.1.x264-EbP
Processing succeeded for Boss.S01E02.720p.HDTV.DD5.1.x264-EbP.nzb

schumi2004 commented:
Strange indeed.
I noticed it a few times also for both movies and series, maybe it's a SABnzbd issue. I'll keep an eye on it as well.

@Tolstyak
Were do i find this Manage failed downloads tab?
I did a clean install but i'm guessing it only shows up if you have failed downloads or not?

Kwebbel commented:
Ok so if its a problem with movies as well it seems like a sabnzb issue. 
Keeps standing the issue with Arrow series not being checked correctly in the history table.

TFenby commented:
Looks like I need to loosen up the comparison in the search a bit.

@schumi2004: Under the manage tab.

schumi2004 commented:
@Tolstyak
Ok, i don't see it here. What i did see was this http://pastebin.com/C41GbJ0L

It happens after i shutdown your failed-database branch and restart SB. When you delete the DB's all is working again.
(everything is a clean setup on my virtual machine)

TFenby commented:
@schumi2004: Ahh, I guess I hadn't pushed up the other changes I made. Grab the latest copy.

@Kwebbel: Try this latest one.

Kwebbel commented:
@Tolstyak: Seems to solve the problem 

Feb-07 18:57:02 INFO     SEARCHQUEUE-BACKLOG-257655 :: Arrow S01E12 720p HDTV X264 DIMENSION has previously failed, rejecting it
Feb-07 18:57:02 DEBUG    SEARCHQUEUE-BACKLOG-257655 :: No result picked.

Another error follows but I think this is the same error posted previously which was said to be not related to failed-handling script.

Exception in thread SEARCHQUEUE-BACKLOG-257655:
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 551, in __bootstrap_inner
    self.run()
  File "/usr/lib/python2.7/threading.py", line 504, in run
    self.__target(_self.__args, *_self.__kwargs)
  File "/opt/Sick-Beard/sickbeard/search_queue.py", line 210, in execute
    search.snatchEpisode(curResult)
  File "/opt/Sick-Beard/sickbeard/search.py", line 108, in snatchEpisode
    if result.resultType in ("nzb", "nzbdata"):
AttributeError: 'NoneType' object has no attribute 'resultType'

schumi2004 commented:
@Kwebbel
Your talking about failed_database branch right ?
Seems to run fine here also, going to merge it with my custom branch

Kwebbel commented:
no i'm talking about the development branch :)

schumi2004 commented:
Ok funny, i'm running failed_db one ;)

Kwebbel commented:
By the way: does anyone have an idea why such a great feature isnt put into the sickbeard master branch? Dont understand the whole point of those pull requests if they never seem to be accepted. Or do I have a wrong idea about how this should work?

simonk83 commented:
Nope.  You're not the first to ask and you won't be the last :).   Things
are pretty tightly controlled in this little corner of the world.   This is
one that should absolutely be given some priority, but if the right couple
of people don't agree then that's the end of that.

On Friday, February 8, 2013, Kwebbel wrote:

> By the way: does anyone have a idea why such a great addition isnt put
> into the sickbeard master branch? Dont understand the whole point of those
> pull requests if they never seem to be accepted. Or do I have a wrong idea
> about how this should work?
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/midgetspy/Sick-Beard/pull/445#issuecomment-13259977.

Kwebbel commented:
So it seems. Great thanks to Tolstyak for writing and maintaining such a great feature even when he knows it might never end up in the sickbeard master branch. I'll be using his anyway because to me it has become a crucial feature. :+1:

simonk83 commented:
I agree, it's pretty much a core feature for me.

On Friday, February 8, 2013, Kwebbel wrote:

> So it seems. Great thanks to Tolstyak for writing and maintaining such a
> great feature even when he knows it might never end up in the sickbeard
> master branch. I'll be using his anyway because to me it has become a
> crucial feature. [image: :+1:]
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/midgetspy/Sick-Beard/pull/445#issuecomment-13262231.

schumi2004 commented:
Same for me , can't live without it anymore since i first used it ;)

Kwebbel commented:
First pull request submitted ;)

https://github.com/Tolstyak/Sick-Beard/pulls

TFenby commented:
They're maintaining, for free, an awesome application that we all make extensive use of. They've earned the right to merge things as they deem appropriate.

Still, see my comment on your pull request.

simonk83 commented:
True, and their approach does work as generally SB is lovely and stable, so
I get it.   There's definitely room for more flexibility with some of the
more core feature pull requests such as this one, even if its just
extensively tested in dev first.

Anyway, I agree, can't complain to much, and we can still use this pull
easily enough ourselves without having to wait, it's just a shame that so
many are missing out on such a basic, yet essential feature.

On Friday, February 8, 2013, Tyler wrote:

> They're maintaining, for free, an awesome application that we all make
> extensive use of. They've earned the right to merge things as they deem
> appropriate.
> 
> Still, see my comment on your pull request.
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/midgetspy/Sick-Beard/pull/445#issuecomment-13264594.

Kwebbel commented:
agreed indeed but imho open-source gets better fast if you share and merge awesome features.

schumi2004 commented:
@Tolstyak
Regarding your failed-database branche.
I got a few failed downloads in it now but the only thing i can do is delete them!
Shouldn't i be able to re queue download or something like that ?

TFenby commented:
That page is just for managing the history of failed downloads, in case you get a new usenet provider or something. Everything else should still be automatic.

schumi2004 commented:
Ah ok, i misunderstood then. I thought i could influence the process there.
Like selecting same release but on a later time something like that, or a selection in choice.

schumi2004 commented:
Why closing it, i presume your opening a new PR ?

TFenby commented:
Apparently github doesn't like when you rename branches.

schumi2004 commented:
Valid answer :P

TFenby commented:
#579
